tests/Avalonia.Skia.UnitTests/Media/TextFormatting/TextLayoutTests.cs

```diff
@@ -575,6 +574,29 @@ public void Should_Wrap()
             }
         }
 
+        [Fact]
+        public void Should_Process_Multiple_NewLines_Properly()
+        {
+            using (Start())
+            {
+                var text = "123\r\n\r\n456\r\n\r\n";
+                var layout = new TextLayout(
+                    text,
+                    Typeface.Default,
+                    12.0f,
+                    Brushes.Black);
+
+                Assert.Equal(4, layout.TextLines.Count);
```

Comment:
That should be 5. When the text ends with a line break an empty line is created.